ci/azure-pipelines-build-test.yml

```diff
@@ -0,0 +1,46 @@
+steps:
+- script: |
+    mkdir -p '$(GOBIN)'
+    mkdir -p '$(GOPATH)/pkg'
+    mkdir -p '$(modulePath)'
+    shopt -s extglob
+    mv !(gopath) '$(modulePath)'
+    echo '##vso[task.prependpath]$(GOBIN)'
+    echo '##vso[task.prependpath]$(GOROOT)/bin'
+  displayName: 'Set up the Go workspace'
+
+ - script: |
+    go version
+    go get -v -t -d ./...
+    if [ -f Gopkg.toml ]; then
+        curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+        dep status
+    fi
+  workingDirectory: '$(modulePath)'
+  displayName: 'Check project dependencies'
+
+ - script: |
+    make build
+  workingDirectory: '$(modulePath)'
+  displayName: 'Build clusterctl, cluster-controller and machine-controller'
+
+ - script: |
+    set -e
+    echo "Running unit tests for machine actuator"
+    go test github.com/platform9/azure-provider/cloud/azure/actuators/machine -test.timeout 0 -v
```

Comment:
Use `make machine-unit-tests`. We shouldn't disable timeout for the unit tests. The default timeout of 10 minutes is long enough at the moment.